### ISSUE TITLE
npm-cli install error

### ISSUE THREAD

**Issue by tarunbatta** [npm/npm#7332]
Make sure you are in the directory you want to scaffold into.
This generator can also be run with: yo webapp

```
     _-----_
    |       |    .--------------------------.
    |--(o)--|    |    Welcome to Yeoman,    |
   `---------´   |   ladies and gentlemen!  |
    ( _´U`_ )    '--------------------------'
    /___A___\
     |  ~  |
   __'.___.'__
 ´   `  |° ´ Y `

Out of the box I include HTML5 Boilerplate, jQuery, and a Gruntfile.js to build your app.
? What more would you like? Bootstrap, Sass, Modernizr
? Would you like to use libsass? Read up more at
https://github.com/andrew/node-sass#node-sass: Yes
   create Gruntfile.js
   create package.json
   create .gitignore
   create .gitattributes
   create .bowerrc
   create bower.json
   create .jshintrc
   create .editorconfig
   create app\styles\main.scss
   create app\favicon.ico
   create app\robots.txt
   create app\index.html
   create app\scripts\main.js
   invoke   mocha


I'm all done. Running bower install & npm install for you to install the required dependencies. If this fails, try running the command your


   create     test\bower.json
   create     test\.bowerrc
   create     test\spec\test.js
   create     test\index.html
bower modernizr#~2.8.2          cached git://github.com/Modernizr/Modernizr.git#2.8.3
bower modernizr#~2.8.2        validate 2.8.3 against git://github.com/Modernizr/Modernizr.git#~2.8.2
bower chai#~1.8.0               cached git://github.com/chaijs/chai.git#1.8.1
bower chai#~1.8.0             validate 1.8.1 against git://github.com/chaijs/chai.git#~1.8.0
bower bootstrap-sass-official#~3.2.0           cached git://github.com/twbs/bootstrap-sass.git#3.2.0+2
bower bootstrap-sass-official#~3.2.0         validate 3.2.0+2 against git://github.com/twbs/bootstrap-sass.git#~3.2.0
bower mocha#~1.14.0             cached git://github.com/mochajs/mocha.git#1.14.0
bower mocha#~1.14.0           validate 1.14.0 against git://github.com/mochajs/mocha.git#~1.14.0
bower mocha#~1.14.0            install mocha#1.14.0
bower chai#~1.8.0              install chai#1.8.1
bower jquery#>= 1.9.0                          cached git://github.com/jquery/jquery.git#2.1.3
bower jquery#>= 1.9.0                        validate 2.1.3 against git://github.com/jquery/jquery.git#>= 1.9.0

mocha#1.14.0 bower_components\mocha

chai#1.8.1 bower_components\chai
bower bootstrap-sass-official#~3.2.0          install bootstrap-sass-official#3.2.0+2
bower modernizr#~2.8.2                        install modernizr#2.8.3
bower jquery#>= 1.9.0                         install jquery#2.1.3

bootstrap-sass-official#3.2.0+2 bower_components\bootstrap-sass-official
└── jquery#2.1.3

modernizr#2.8.3 bower_components\modernizr

jquery#2.1.3 bower_components\jquery
npm WARN package.json testapp3@ No description
npm WARN package.json testapp3@ No repository field.
npm WARN package.json testapp3@ No README data
npm ERR! Windows_NT 6.1.7601
npm ERR! argv "C:\\Program Files\\nodejs\\\\node.exe" "C:\\Program Files\\nodejs\\node_modules\\npm\\bin\\npm-cli.js" "install"
npm ERR! node v0.12.0
npm ERR! npm  v2.5.1

npm ERR! null == true
npm ERR!
npm ERR! If you need help, you may report this error at:
npm ERR!     <http://github.com/npm/npm/issues>
\
> phantomjs@1.9.15 install C:\Users\tarunb\Data\testApp3\node_modules\grunt-mocha\node_modules\grunt-lib-phantomjs\node_modules\phantomjs
> node install.js
```

**Comment by smikes** [comment]
Can you please upload the `package.json` from the `testapp3` as a gist <https://gist.github.com> ?

**Comment by tarunbatta** [comment]
Hi smikes,

This app is created via following command,
yo webapp

The following is the package.json,
////////////////////////////////////
{
  "name": "testapp3",
  "devDependencies": {
    "apache-server-configs": "^2.7.1",
    "grunt": "^0.4.5",
    "grunt-autoprefixer": "^1.0.0",
    "grunt-concurrent": "^0.5.0",
    "grunt-contrib-clean": "^0.6.0",
    "grunt-contrib-concat": "^0.5.0",
    "grunt-contrib-connect": "^0.8.0",
    "grunt-contrib-copy": "^0.5.0",
    "grunt-contrib-cssmin": "^0.10.0",
    "grunt-contrib-htmlmin": "^0.3.0",
    "grunt-contrib-imagemin": "^0.8.0",
    "grunt-contrib-jshint": "^0.10.0",
    "grunt-contrib-uglify": "^0.5.1",
    "grunt-contrib-watch": "^0.6.1",
    "grunt-mocha": "^0.4.10",
    "grunt-modernizr": "^0.5.2",
    "grunt-newer": "^0.7.0",
    "grunt-rev": "^0.1.0",
    "grunt-sass": "^0.14.0",
    "grunt-svgmin": "^0.4.0",
    "grunt-usemin": "^2.3.0",
    "grunt-wiredep": "^1.7.0",
    "jshint-stylish": "^0.4.0",
    "load-grunt-tasks": "^0.4.0",
    "time-grunt": "^0.4.0"
  },
  "engines": {
    "node": ">=0.10.0"
  }
}
////////////////////////////////////

Regards
Tarun

**Comment by tarunbatta** [comment]
https://gist.github.com/9604f20d6df80769e21b.git

**Comment by smikes** [comment]
You are using the newly-released `v0.12.0` version of node. You may have to downgrade to `v0.10.36` if some of the dependencies you're installing have not been updated to support 0.12

The first thing I'd recommend trying is installing PhantomJS for Windows separately: http://phantomjs.org/download.html

**Comment by othiym23** [comment]
```
npm ERR! null == true
```

This unfortunately terse assertion message generally means that something bad has happened inside Node itself, especially when unaccompanied by a stacktrace. When I've seen it before, it's been an OpenSSL error of some kind or another, but it could be caused by almost anything, and without a stack trace, it's nearly impossible to resolve.

tl;dr if you see that message more than once in a blue moon, you're probably encountering a Node issue, and should work with the Node / io.js core teams to figure out what's going on.

**Comment by wiseguy12851** [comment]
I'm having the same problem. At first I tried it on my Windows laptop and thought it was just some configuration or setup problem or something. When the same error and all happened on my Windows Desktop I decided to poke around Google for some answers and came across this thread.

I think its just a Node error on Windows, Linux and Windows are running version 12 and I think a fresh "npm install" on Linux worked flawlessly but fails without any kind of usable error both times on both Windows.

**Comment by othiym23** [comment]
@wiseguy12851 It may be an npm error, but if it is, it's still ultimately coming from Node, because npm is incapable of emitting that kind of bare assertion message without a stacktrace itself. I believe that there are issues with npm or Node on Windows with Node 0.12, but right now I don't have enough information to even open an issue with the Node core team. So if you have a repro case you can share, that would be helpful.

**Comment by othiym23** [comment]
See #7349 for further discussion of this, as I'm going to close this issue and use that one as the master issue for tracking this problem, which appears to be in Node.js's HTTPS handling code.
